src/runtime_src/core/tools/common/ReportCu.cpp

```diff
@@ -0,0 +1,131 @@
+  boost::property_tree::ptree ptSt_arr;
+  for (int i = 0; i < (int)result.size(); i++)
+    ptSt_arr.push_back(std::make_pair("", boost::property_tree::ptree(result[i])));
+
+  pt.add_child( std::string("bits_set"), ptSt_arr);
+
+  return pt;
+}
+
+boost::property_tree::ptree
+populate_cus(const xrt_core::device *device)
+{
+  boost::property_tree::ptree pt;
+  std::vector<char> ip_buf;
+  std::vector<std::tuple<uint64_t, uint32_t, uint32_t>> cu_stats;
+
+  try {
+    ip_buf = xrt_core::device_query<qr::ip_layout_raw>(device);
+    cu_stats = xrt_core::device_query<qr::kds_cu_info>(device);
+  } catch (const std::exception& ex){
+    pt.put("error_msg", ex.what());
+  }
+
+  if(ip_buf.empty() || cu_stats.empty())
+    return pt;
+
+  const ip_layout *layout = (ip_layout *)ip_buf.data();
+  for (int i = 0; i < layout->m_count; i++) {
+    if (layout->m_ip_data[i].m_type != IP_KERNEL)
+      continue;
+
+    for(auto& stat : cu_stats) {
+      if (layout->m_ip_data[i].m_base_address == (uint64_t)std::get<0>(stat)) {
+        uint32_t usage = std::get<1>(stat);
+        uint32_t status = std::get<2>(stat);
+
+        boost::property_tree::ptree ptCu;
+        ptCu.put( "name",			layout->m_ip_data[i].m_name);
+        ptCu.put( "base_address",		boost::str(boost::format("0x%x") % std::get<0>(stat)));
+        ptCu.put( "usage",			usage);
+        ptCu.add_child( std::string("status"),  parse_cu_status(status));
+        pt.push_back(std::make_pair("", ptCu));
+      }
+    }
+  }
+
+  return pt;
+}
+
+void
+ReportCu::getPropertyTreeInternal(const xrt_core::device * _pDevice, 
+                                              boost::property_tree::ptree &_pt) const
+{
+  // Defer to the 20202 format.  If we ever need to update JSON data, 
+  // Then update this method to do so.
+  getPropertyTree20202(_pDevice, _pt);
+}
+
+void 
+ReportCu::getPropertyTree20202( const xrt_core::device * _pDevice, 
+                                           boost::property_tree::ptree &_pt) const
+{
+  // There can only be 1 root node
+  _pt.add_child("compute_units", populate_cus(_pDevice));
+}
+
+void 
+ReportCu::writeReport( const xrt_core::device * _pDevice,
+                                  const std::vector<std::string> & /*_elementsFilter*/, 
+                                  std::iostream & _output) const
+{
+  boost::property_tree::ptree _pt;
+  boost::property_tree::ptree empty_ptree;
+  getPropertyTreeInternal(_pDevice, _pt);
+
+  _output << boost::format("    %-8s%-24s%-16s%-8s%-8s\n") % "Index" % "Name" % "Base_Address" % "Usage" % "Status";
+  try {
+    int index = 0;
+    boost::property_tree::ptree& v = _pt.get_child("compute_units");
+    for(auto& kv : v) {
+      boost::property_tree::ptree& cu = kv.second;
+      std::string cu_status = cu.get_child("status").get<std::string>("bit_mask");
+      uint32_t status_val = std::stoul(cu_status, nullptr, 16);
+      _output << boost::format("    %-8s%-24s%-16s%-8s%-8s\n") % index++ %

```

Comment:
Being that this boost::format statement is the same as the one on line 113, please create ONLY one boost formating signature and re-use it for both.  For example:
```
boost::format cuFmt("    %-8s%-24s%-16s%-8s%-8s\n");
_output << cuFmt %  % index++ %
	      cu.get<std::string>("name") % cu.get<std::string>("base_address") %
	      cu.get<std::string>("usage") % xrt_core::utils::parse_cu_status(status_val);
```